### PR TITLE
PHOENIX-6633 Use standard JDBC URL in Spark Connector and make it opt…

### PR DESCRIPTION
**Pull request by stoty**
…ional

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 46s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |  10m 52s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 37s |  master passed  |
| +1 :green_heart: |  compile  |   1m 26s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 31s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 50s |  phoenix5-spark in master has 40 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   0m 40s |  phoenix5-spark3 in master has 22 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  8s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 26s |  the patch passed  |
| -1 :x: |  javac  |   0m 29s |  phoenix5-spark generated 54 new + 105 unchanged - 46 fixed = 159 total (was 151)  |
| -1 :x: |  javac  |   0m 29s |  phoenix5-spark3 generated 48 new + 141 unchanged - 45 fixed = 189 total (was 186)  |
| -1 :x: |  checkstyle  |   0m  8s |  phoenix5-spark: The patch generated 12 new + 74 unchanged - 4 fixed = 86 total (was 78)  |
| -1 :x: |  checkstyle  |   0m  8s |  phoenix5-spark3: The patch generated 6 new + 86 unchanged - 3 fixed = 92 total (was 89)  |
| -1 :x: |  markdownlint  |   0m  2s |  The patch generated 14 new + 92 unchanged - 13 fixed = 106 total (was 105)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 6 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  javadoc  |   0m 11s |  phoenix5-spark generated 9 new + 155 unchanged - 1 fixed = 164 total (was 156)  |
| -1 :x: |  javadoc  |   0m 11s |  phoenix5-spark3 generated 6 new + 188 unchanged - 3 fixed = 194 total (was 191)  |
| +1 :green_heart: |  spotbugs  |   1m 40s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 45s |  phoenix5-spark in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 39s |  phoenix5-spark3 in the patch passed.  |
| +1 :green_heart: |  unit  |  25m  2s |  phoenix5-spark3-it in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 19s |  The patch does not generate ASF License warnings.  |
|  |   |  66m 30s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/108 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile markdownlint |
| uname | Linux 8575f3bc1b0c 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / ca76ff8 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/diff-compile-javac-phoenix5-spark.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/diff-compile-javac-phoenix5-spark3.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix5-spark.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix5-spark3.txt |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/whitespace-eol.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark3.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/testReport/ |
| Max. process+thread count | 1934 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark phoenix5-spark3 phoenix5-spark3-it U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 15s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |  10m 47s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m  6s |  master passed  |
| +1 :green_heart: |  compile  |   1m 27s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 16s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 59s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 55s |  phoenix5-spark in master has 40 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   0m 42s |  phoenix5-spark3 in master has 22 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   1m 42s |  root in master has 111 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 24s |  the patch passed  |
| -1 :x: |  javac  |   1m 24s |  root generated 103 new + 2032 unchanged - 92 fixed = 2135 total (was 2124)  |
| -1 :x: |  checkstyle  |   0m 16s |  root: The patch generated 12 new + 166 unchanged - 1 fixed = 178 total (was 167)  |
| -1 :x: |  markdownlint  |   0m  3s |  The patch generated 14 new + 92 unchanged - 13 fixed = 106 total (was 105)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 7 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 12s |  phoenix5-spark generated 12 new + 152 unchanged - 4 fixed = 164 total (was 156)  |
| -1 :x: |  javadoc  |   0m 11s |  phoenix5-spark3 generated 6 new + 188 unchanged - 3 fixed = 194 total (was 191)  |
| -1 :x: |  javadoc  |   0m 27s |  root generated 39 new + 2109 unchanged - 28 fixed = 2148 total (was 2137)  |
| +1 :green_heart: |  spotbugs  |   3m 36s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  54m 58s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 26s |  The patch generated 3 ASF License warnings.  |
|  |   |  99m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/108 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile markdownlint xml |
| uname | Linux fa79d20b35f9 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / ca76ff8 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/whitespace-eol.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark3.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/diff-javadoc-javadoc-root.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 2057 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark phoenix5-spark3 phoenix5-spark3-it . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/2/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |  10m 55s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  10m 57s |  master passed  |
| +1 :green_heart: |  compile  |   1m 21s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 16s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  4s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 59s |  phoenix5-spark in master has 40 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   0m 44s |  phoenix5-spark3 in master has 22 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   2m  1s |  root in master has 111 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  10m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 27s |  the patch passed  |
| -1 :x: |  javac  |   1m 27s |  root generated 97 new + 2020 unchanged - 104 fixed = 2117 total (was 2124)  |
| -1 :x: |  checkstyle  |   0m 16s |  root: The patch generated 2 new + 158 unchanged - 9 fixed = 160 total (was 167)  |
| -1 :x: |  markdownlint  |   0m  3s |  The patch generated 10 new + 91 unchanged - 14 fixed = 101 total (was 105)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 2 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 11s |  phoenix5-spark generated 1 new + 151 unchanged - 5 fixed = 152 total (was 156)  |
| -1 :x: |  javadoc  |   0m 13s |  phoenix5-spark3 generated 1 new + 187 unchanged - 4 fixed = 188 total (was 191)  |
| -1 :x: |  javadoc  |   0m 28s |  root generated 2 new + 2128 unchanged - 9 fixed = 2130 total (was 2137)  |
| +1 :green_heart: |  spotbugs  |   4m  2s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  59m  6s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 27s |  The patch generated 3 ASF License warnings.  |
|  |   | 106m 15s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/108 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile markdownlint xml |
| uname | Linux 87a47ed9a953 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / ca76ff8 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/whitespace-eol.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark3.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/diff-javadoc-javadoc-root.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 1967 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark phoenix5-spark3 phoenix5-spark3-it . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix5-spark/src/main/java/org/apache/phoenix/spark/datasource/v2/PhoenixDataSource.java [hunk 1]
```diff
@@ -42,6 +45,7 @@ public class PhoenixDataSource  implements DataSourceV2,  ReadSupport, WriteSupp
     private static final Logger logger = LoggerFactory.getLogger(PhoenixDataSource.class);
     public static final String SKIP_NORMALIZING_IDENTIFIER = "skipNormalizingIdentifier";
     public static final String ZOOKEEPER_URL = "zkUrl";
```
Comment (richardantal) [inline phoenix5-spark/src/main/java/org/apache/phoenix/spark/datasource/v2/PhoenixDataSource.java:47]: You added a Deprecated tag to the ZOOKEEPER_URL in phoenix5-spark3/src/main/java/org/apache/phoenix/spark/sql/connector/PhoenixDataSource.java 
why not here too?

File: phoenix5-spark3/src/main/java/org/apache/phoenix/spark/sql/connector/writer/PhoenixDataSourceWriteOptions.java [hunk 2]
```diff
@@ -28,20 +28,20 @@
...
 
-    private PhoenixDataSourceWriteOptions(String tableName, String zkUrl, String scn,
+    private PhoenixDataSourceWriteOptions(String tableName, String jdbcUrl, String scn,
                                           String tenantId, StructType schema, boolean skipNormalizingIdentifier,
                                           Properties overriddenProps) {
         if (tableName == null) {
             throw new NullPointerException();
         }
-        if (zkUrl == null) {
+        if (jdbcUrl == null) {
```
Comment (richardantal) [inline phoenix5-spark3/src/main/java/org/apache/phoenix/spark/sql/connector/writer/PhoenixDataSourceWriteOptions.java:44]: Why don't you throw IllegalArgumentException if the schema or the overriddenPropsos null?
Like it is done int the phoenix5-spark connector

**Review by richardantal — COMMENTED** [review]
Looks Good I left some minor comments

File: phoenix5-spark3/src/main/java/org/apache/phoenix/spark/sql/connector/writer/PhoenixDataSourceWriteOptions.java (hunk 2 above)
Comment (stoty) [reply phoenix5-spark3/src/main/java/org/apache/phoenix/spark/sql/connector/writer/PhoenixDataSourceWriteOptions.java:44]: Corrected.

File: phoenix5-spark/src/main/java/org/apache/phoenix/spark/datasource/v2/PhoenixDataSource.java (hunk 1 above)
Comment (stoty) [reply phoenix5-spark/src/main/java/org/apache/phoenix/spark/datasource/v2/PhoenixDataSource.java:48]: Corrected.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |  11m  4s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 40s |  master passed  |
| +1 :green_heart: |  compile  |   1m 17s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 15s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 58s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 53s |  phoenix5-spark in master has 40 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   0m 41s |  phoenix5-spark3 in master has 22 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   1m 39s |  root in master has 111 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 27s |  the patch passed  |
| -1 :x: |  javac  |   1m 27s |  root generated 118 new + 1999 unchanged - 125 fixed = 2117 total (was 2124)  |
| -1 :x: |  checkstyle  |   0m 15s |  root: The patch generated 2 new + 158 unchanged - 9 fixed = 160 total (was 167)  |
| -1 :x: |  markdownlint  |   0m  4s |  The patch generated 10 new + 91 unchanged - 14 fixed = 101 total (was 105)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 2 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 12s |  phoenix5-spark generated 1 new + 151 unchanged - 5 fixed = 152 total (was 156)  |
| -1 :x: |  javadoc  |   0m 11s |  phoenix5-spark3 generated 1 new + 187 unchanged - 4 fixed = 188 total (was 191)  |
| -1 :x: |  javadoc  |   0m 27s |  root generated 2 new + 2128 unchanged - 9 fixed = 2130 total (was 2137)  |
| +1 :green_heart: |  spotbugs  |   3m 35s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  55m 28s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 26s |  The patch generated 3 ASF License warnings.  |
|  |   |  97m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/108 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile markdownlint xml |
| uname | Linux 6703f5426617 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / ca76ff8 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/whitespace-eol.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix5-spark3.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/diff-javadoc-javadoc-root.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 1958 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark phoenix5-spark3 phoenix5-spark3-it . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-108/4/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by richardantal — APPROVED** [review]
LGTM

**Pull request closed without merging**